### PR TITLE
tests: make pip_parse example work with bzlmod enabled

### PR DESCRIPTION
**Pull request by rickeylev**
Bazel is enabling bzlmod by default, which means the examples need to be updated to be bzlmod compatible.

Work towards #1520

### CODE REVIEW

File: examples/pip_parse/BUILD.bazel [hunk 1]
```diff
@@ -50,9 +45,9 @@ py_test(
 # For pip dependencies which have entry points, the `entry_point` macro can be
 # used from the generated `pip_parse` repository to access a runnable binary.
 
-alias(
+py_console_script_binary(
     name = "yamllint",
-    actual = entry_point("yamllint"),
+    pkg = "@pypi//yamllint:pkg",
```
Comment (aignas) [inline examples/pip_parse/BUILD.bazel:50]: nit:
```suggestion
    pkg = "@pypi//yamllint",
```

File: examples/pip_parse/WORKSPACE [hunk 2]
```diff
@@ -24,6 +24,7 @@ pip_parse(
     # can be passed
     # environment = {"HTTPS_PROXY": "http://my.proxy.fun/"},
     name = "pypi",
+    incompatible_generate_aliases = True,
```
Comment (aignas) [inline examples/pip_parse/WORKSPACE:27]: nit: Please wait for #1525 and remove this flag.

File: examples/pip_parse/pip_parse_test.py [hunk 3]
```diff
@@ -19,20 +19,28 @@
...
 
 class PipInstallTest(unittest.TestCase):
     maxDiff = None
 
+    def _remove_leading_dirs(self, paths):
+        # Removes the first two directories (external/<reponame>)
+        # to normalize what workspace and bzlmod produce.
+        #return [str(Path(*Path(v).parts[2:])) for v in paths]
+        return [
+            '/'.join(v.split('/')[2:])
```
Comment (aignas) [inline examples/pip_parse/pip_parse_test.py:33]: FYI, I just wanted to comment that we should use an os specific path separator, but realized that the values come from `rlocationpaths` via an env var which will always use `/` separators, right?

**Review by aignas — APPROVED** [review]

File: examples/pip_parse/WORKSPACE (hunk 2 above)
Comment (rickeylev) [reply examples/pip_parse/WORKSPACE:27]: Done

File: examples/pip_parse/BUILD.bazel (hunk 1 above)
Comment (rickeylev) [reply examples/pip_parse/BUILD.bazel:50]: Done

File: examples/pip_parse/pip_parse_test.py (hunk 3 above)
Comment (rickeylev) [reply examples/pip_parse/pip_parse_test.py:33]: Correct. Originally I used pathlib.Path to do this, but on windows, it changed things to `\` and then the paths didn't match the expected values.

**Pull request merged**